src/modm/board/thingplus_rp2040/module.lb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2022, Andrey Kunitsyn
+# Copyright (c) 2022, Nikolay Semenov
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+def init(module):
+    module.name = ":board:thingplus-rp2040"
+    module.description = """\
+# SparkFun Thing Plus - RP2040
+
+The SparkFun Thing Plus - RP2040 is a low-cost, high performance board
+with flexible digital interfaces featuring the Raspberry Pi Foundation's RP2040 microcontroller.
+Besides the Thing Plus or Feather footprint (with 18 GPIO pins),
+the board also includes an SD card slot, 16MB (128Mbit) flash memory,
+a JST single cell battery connector (with a charging circuit and fuel gauge sensor),
+an addressable WS2812 RGB LED, JTAG PTH pins, four (4-40 screw) mounting holes,
+and SparkFun's Qwiic connector.
+
+https://www.sparkfun.com/products/17745
+
+See `modm:rp-pico` for programming instructions.
+"""
+
+def prepare(module, options):
+    if not options[":target"].partname.startswith("rp2040"):
+        return False
+
+    module.depends(
+        ":platform:clock",
+        ":platform:core",
+        ":platform:gpio",
+        ":platform:usb",
+        ":platform:clockgen")
+    return True
+
+def build(env):
+    env.outbasepath = "modm/src/modm/board"
+    env.substitutions = {
+        "with_logger": False,
+        "with_assert": env.has_module(":architecture:assert")
+    }
+    env.template("../board.cpp.in", "board.cpp")
+    env.copy('.')
+
+    env.outbasepath = "modm/openocd/modm/board/"
+    env.copy(repopath("tools/openocd/modm/rp2040_picoprobe.cfg"), "rp2040_picoprobe.cfg")
+    env.collect(":build:openocd.source", "modm/board/rp2040_picoprobe.cfg")
```

Comment:
What about moving the programmer configuration into its own module, since it is not part of the actual devboard?

The RP2040-Picoprobe module can still be included in the blink example in `project.xml`.

(This should of course also apply to all existing (STM32) Blue-Pill/Black-Pill/... Dev-Board without integrated programmer, but that's out of the scope of this pull request.)

Reply:
Not super convinved, since it's just a single file, so seems weird to have it's own module for it.
Maybe it would make more if we had some "debug" module that's more separate from the build module, but that's kinda weird too.
If you write your own non-BSP based application, then you need to copy this file manually anyways?

Reply:
I always found it annoying in the past with boards like the Blue-Pill to have a different programmer (e.g. stlink v2 vs. stlink v3) than the one assumed in the BSP for no reason and then have to manually patch around in the generated code each time.

> it's just a single file, so seems weird to have it's own module for it.

I do not see any disadvantage?
Probably the module should also add the logging functionality to the board (if the programmer has a integrated USB VCP channel) and then document the necessary wiring.

>  Maybe it would make more if we had some "debug" module that's more separate from the build module, but that's kinda weird too.

[`modm:debug` module](https://modm.io/reference/module/modm-debug/) exists, and contains `modm::log::Logger`. Why is it called `debug`?

> If you write your own non-BSP based application, then you need to copy this file manually anyways?

Or just include the programmer module, if that is a module?

Reply:
> have to manually patch around in the generated code each time.

Yeah, that's really annoying. But does specifying the programmer in the `openocd.cfg` file  not work as documented in the Blue Pill docs?

> Probably the module should also add the logging functionality to the board (if the programmer has a integrated USB VCP channel) and then document the necessary wiring.

Ok, I like that.

> [modm:debug module](https://modm.io/reference/module/modm-debug/) exists, and contains modm::log::Logger. Why is it called debug?

Good old historic reasons of course. I would move it to `modm:io:logger`.

> Or just include the programmer module, if that is a module?

Sure, but if the BSP already includes the programmer module, then you cannot uninclude it again. So you're back to square one?

Reply:
I'm not opposed to this at all btw, I think it's a really good idea to separate the programmer out, it just needs to be done well as per usual ;-P